src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/perform_interaction_request.cc

```diff
@@ -1063,13 +1092,68 @@ void PerformInteractionRequest::SendBothModeResponse(
       msg_param.empty() ? NULL : &msg_param;
   std::string info = app_mngr::commands::MergeInfos(
       ui_perform_info, ui_info_, vr_perform_info, vr_info_);
+
   DisablePerformInteraction();
+
   SendResponse(result,
                perform_interaction_result_code,
                info.empty() ? NULL : info.c_str(),
                response_params);
 }
 
+mobile_apis::Result::eType
+PerformInteractionRequest::PrepareResultCodeForResponse(
+    const app_mngr::commands::ResponseInfo& ui_response,
+    const app_mngr::commands::ResponseInfo& vr_response) {
+  if (mobile_apis::InteractionMode::VR_ONLY == interaction_mode_) {
+    if (FirstAnsweredInterface::VR == first_responser_) {
+      return MessageHelper::HMIToMobileResult(vr_result_code_);
```

Comment:
@YarikMamykin As in the method we have so many `if` statements - may be it is worth to add short logging to every `if`?